File: app/src/main/java/com/savvasdalkitsis/gameframe/widget/view/NextWidgetProvider.java
package com.savvasdalkitsis.gameframe.widget.view;

import com.savvasdalkitsis.gameframe.R;

public class NextWidgetProvider extends ClickableWidgetProvider {

    @Override
    protected int layoutResId() {
        return R.layout.widget_next;
    }

    @Override
    protected void onClick() {
        presenter.next();
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/feature/widget/view/NextWidgetProvider.kt
package com.savvasdalkitsis.gameframe.feature.widget.view

import com.savvasdalkitsis.gameframe.R

class NextWidgetProvider : ClickableWidgetProvider() {

    override fun layoutResId() = R.layout.widget_next

    override fun onClick() {
        presenter.next()
    }
}
